File: samples/dotnet/BgeM3.Onnx/ExecutionProvider.cs
namespace BgeM3.Onnx;

/// <summary>
/// Supported execution providers for ONNX inference
/// </summary>
public enum ExecutionProvider
{
    /// <summary>
    /// CPU execution provider (default, always available)
    /// </summary>
    CPU,

    /// <summary>
    /// CUDA execution provider (requires CUDA-enabled GPU and CUDA runtime)
    /// </summary>
    CUDA
}

File: samples/java/bge-m3-onnx/src/main/java/com/yunikosoftware/bgem3onnx/ExecutionProvider.java
package com.yunikosoftware.bgem3onnx;

/**
 * Supported execution providers for ONNX inference
 */
public enum ExecutionProvider {
    /**
     * CPU execution provider (default, always available)
     */
    CPU,

    /**
     * CUDA execution provider (requires CUDA-enabled GPU and CUDA runtime)
     */
    CUDA
}